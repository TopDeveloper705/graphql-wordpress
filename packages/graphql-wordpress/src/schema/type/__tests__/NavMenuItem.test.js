import NavMenuItemType from 'type/NavMenuItem';

// eslint-disable-next-line no-underscore-dangle
const fields = NavMenuItemType._typeConfig.fields;

describe('Test NavMenuItem type', () => {
  test('Test name', () => {
    expect(NavMenuItemType.name).toMatchSnapshot();
  });

  test('Test fields', () => {
    expect(Object.keys(NavMenuItemType.getFields())).toMatchSnapshot();
  });

  test('Test resolve id', () => {
    const id = fields.id.resolve({ id: 13 });
    expect(id).toMatchSnapshot();
  });

  test('Test resolve parent', () => {
    const parent = fields.parent.resolve({ parent: 13 });
    expect(parent).toMatchSnapshot();
  });

  test('Test resolve parent null', () => {
    const parent = fields.parent.resolve({ parent: 0 });
    expect(parent).toMatchSnapshot();
  });

  test('Test resolve typeName', () => {
    const typeName = fields.typeName.resolve({ type_name: 'category' });
    expect(typeName).toMatchSnapshot();
  });

  test('Test resolve typeSlug', () => {
    const typeSlug = fields.typeSlug.resolve({ type_slug: 'music' });
    expect(typeSlug).toMatchSnapshot();
  });

  test('Test resolve dataSlug', () => {
    const dataSlug = fields.dataSlug.resolve({ object_slug: 'watch-this' });
    expect(dataSlug).toMatchSnapshot();
  });

  test('Test resolve dataID', () => {
    const dataID = fields.dataID.resolve({ object_id: 4 });
    expect(dataID).toMatchSnapshot();
  });

  test('Test resolve dataID null', () => {
    const dataID = fields.dataID.resolve({ object_id: 0 });
    expect(dataID).toMatchSnapshot();
  });
});
