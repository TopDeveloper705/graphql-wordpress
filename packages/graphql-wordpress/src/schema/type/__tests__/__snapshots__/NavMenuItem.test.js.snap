// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NavMenuItem type Test fields 1`] = `
Array [
  "id",
  "parent",
  "order",
  "type",
  "typeName",
  "typeSlug",
  "dataSlug",
  "dataID",
  "url",
  "title",
  "target",
]
`;

exports[`Test NavMenuItem type Test name 1`] = `"NavMenuItem"`;

exports[`Test NavMenuItem type Test resolve dataID 1`] = `"TmF2TWVudUl0ZW06NA=="`;

exports[`Test NavMenuItem type Test resolve dataID null 1`] = `null`;

exports[`Test NavMenuItem type Test resolve dataSlug 1`] = `"watch-this"`;

exports[`Test NavMenuItem type Test resolve id 1`] = `"TmF2TWVudUl0ZW06MTM="`;

exports[`Test NavMenuItem type Test resolve parent 1`] = `"TmF2TWVudUl0ZW06MTM="`;

exports[`Test NavMenuItem type Test resolve parent null 1`] = `null`;

exports[`Test NavMenuItem type Test resolve typeName 1`] = `"category"`;

exports[`Test NavMenuItem type Test resolve typeSlug 1`] = `"music"`;
